### DISCUSSION TITLE
CORS errors during development if the code throws an exception. (example: 500-error)

### DISCUSSION THREAD

**Discussion by engenes** [laravel/framework#46365] [Q&A]
- Laravel Version: 10.2.0
- PHP Version:  8.1.0 

### Description:
I still get the  CORS-error when an 500-error occurs on the server.
For example, if I make a syntax error in the code, causing the application to crash, then the headers will not be set and the browser will show CORS errors.

That is, code execution does not reach ` \Illuminate\Http\Middleware\HandleCors::class` in `app/Http/Kernel.php` if there is a 500-error

Installed clean Laravel and intentionally made a mistake in the code, and again I see the CORS-error, instead of an error message.

It is no longer possible to quickly identify the source of an error in the console during development SPA

I have to open the log file all the time, without normal highlighting

![image](https://user-images.githubusercontent.com/38880660/223036925-04da22b3-643a-4a6f-b571-60529cd9935a.png)

### Steps To Reproduce:

1. install Laravel
2. configure CORS for your client domain
3. create a route

```
Route::get('/test', function () {
     return 111;
});
```
 in routes/api.php

4. send a request from your client domain to this route
5. make sure everything works and there are no CORS errors
6. make a syntax error in routes/api.php and run the request again

result: CORS-error
expect: standard Laravel syntax error message

**Comment by JagdishPrecise** [comment]
By default, Laravel returns a standard syntax error message when an exception occurs. However, if you're encountering CORS errors instead, you can update your exception handler to return the standard Laravel syntax error message instead. This will make debugging your code much easier and prevent CORS errors from disrupting your development workflow.

1. Open your app/Exceptions/Handler.php file.
2. Locate the render method.
3. Remove or comment out any code that is causing the CORS error, such as headers that allow CORS.
4. Add the following code to the render method:

```
if ($exception instanceof \Symfony\Component\Debug\Exception\FatalThrowableError) {
    return response()->json(['message' => 'Server Error'], 500);
}
```
This code will catch any syntax errors that are causing your code to crash, and return a standard Laravel JSON error message instead of a CORS error.

Save the file and test your code again.
With this change, you should now see the standard Laravel syntax error message instead of a CORS error when your code throws an exception.

**Reply by engenes** [reply]
Laravel used to return a standard syntax error message, but as of version 9.2 after Fruitcake/laravel-cors is deprecated, cors error is returned instead of the error message.

I have stock Laravel 10, and the file `app/Exceptions/Handler.php ` looks like this
I think developers should look into this and fall back to the default action when an exception occurs.
```
<?php

namespace App\Exceptions;

use Illuminate\Foundation\Exceptions\Handler as ExceptionHandler;
use Throwable;

class Handler extends ExceptionHandler
{
    /**
     * A list of exception types with their corresponding custom log levels.
     *
     * @var array<class-string<\Throwable>, \Psr\Log\LogLevel::*>
     */
    protected $levels = [
        //
    ];

    /**
     * A list of the exception types that are not reported.
     *
     * @var array<int, class-string<\Throwable>>
     */
    protected $dontReport = [
        //
    ];

    /**
     * A list of the inputs that are never flashed to the session on validation exceptions.
     *
     * @var array<int, string>
     */
    protected $dontFlash = [
        'current_password',
        'password',
        'password_confirmation',
    ];

    /**
     * Register the exception handling callbacks for the application.
     */
    public function register(): void
    {
        $this->reportable(function (Throwable $e) {
            //
        });
    }
}
```
